src/Okta.Idx.Sdk/IdxClient.cs

```diff
@@ -594,12 +595,15 @@ private async Task<AuthenticationResponse> AuthenticateWithPasswordAsync(Authent
                     }
                     else if (challengeResponse.ContainsRemediationOption(RemediationType.SelectAuthenticatorAuthenticate))
                     {
-                        return new AuthenticationResponse
+                        var authenticationResponse = new AuthenticationResponse
                         {
                             IdxContext = idxContext,
                             AuthenticationStatus = AuthenticationStatus.AwaitingChallengeAuthenticatorSelection,
                             Authenticators = IdxResponseHelper.ConvertToAuthenticators(challengeResponse.Authenticators.Value, challengeResponse.AuthenticatorEnrollments.Value),
                         };
+
+                        authenticationResponse.OktaVerifyAuthenticationOptions = new OktaVerifyAuthenticationOptions(authenticationResponse, challengeResponse);

```

Comment:
We should follow the current design where all the information of a specific authenticator is accessed via either CurrentAuthenticator or CurrentAuthenticatorEnrollment.

Reply:
I considered that, but because of the complexity of the options available to Okta Verify it became unwieldy to track and almost impossible to debug.